frontend/app/src/components/inputs/TagInput.vue

```diff
@@ -101,114 +107,114 @@ const valueValidator = (value: any) => {
-    }
-  }
+  setup(props, { emit }) {
+    const { value } = toRefs(props);
+    const { tags, addTag } = setupTags();
 
-  get newTagBackground(): string {
-    return `#${this.colorScheme.backgroundColor}`;
-  }
+    const manageTags = ref<boolean>(false);
 
-  get newTagForeground(): string {
-    return `#${this.colorScheme.foregroundColor}`;
-  }
+    const search = ref<string>('');
 
-  get values(): Tag[] {
-    return this.tags.filter(({ name }) => this.value.includes(name));
-  }
+    const randomScheme = () => {
+      const backgroundColor = randomColor();
+      return {
+        backgroundColor,
+        foregroundColor: invertColor(backgroundColor)
+      };
+    };
 
-  filter(tag: Tag, queryText: string): boolean {
-    const { name, description } = tag;
-    const query = queryText.toLocaleLowerCase();
-    return (
-      name.toLocaleLowerCase().indexOf(query) > -1 ||
-      description.toLocaleLowerCase().indexOf(query) > -1
-    );
-  }
+    const colorScheme = ref(randomScheme());
 
-  remove(tag: string) {
-    const tags = this.value;
-    const index = tags.indexOf(tag);
-    this.input([...tags.slice(0, index), ...tags.slice(index + 1)]);
-  }
+    const tagExists = (tagName: string): boolean => {
+      return tags.value.map(({ name }) => name).includes(tagName);
+    };
 
-  randomScheme() {
-    const backgroundColor = randomColor();
-    return {
-      backgroundColor: backgroundColor,
-      foregroundColor: invertColor(backgroundColor)
+    const createTag = async (name: string) => {
+      const { backgroundColor, foregroundColor } = colorScheme.value;
+      const tag: Tag = {
+        name,
+        description: '',
+        backgroundColor,
+        foregroundColor
+      };
+      return await addTag(tag);
     };
-  }
 
-  tagExists(tagName: string): boolean {
-    return this.tags.map(({ name }) => name).includes(tagName);
-  }
+    const remove = (tag: string) => {
+      const tags = value.value;
+      const index = tags.indexOf(tag);
+      input([...tags.slice(0, index), ...tags.slice(index + 1)]);
+    };
 
-  async createTag(name: string) {
-    const { backgroundColor, foregroundColor } = this.colorScheme;
-    const tag: Tag = {
-      name,
-      description: '',
-      backgroundColor,
-      foregroundColor
+    const attemptTagCreation = (element: string) => {
+      if (tagExists(element)) {
+        return;
+      }
+      createTag(element).then(({ success }) => {
+        if (!success) {
+          remove(element);
+        }
+      });
     };
-    return await this.addTag(tag);
-  }
 
-  @Emit()
-  input(_value: (string | Tag)[]): string[] {
-    const tags: string[] = [];
-    for (let i = 0; i < _value.length; i++) {
-      const element = _value[i];
-      if (typeof element === 'string') {
-        this.attemptTagCreation(element);
-        tags.push(element);
-      } else {
-        tags.push(element.name);
+    const input = (_value: (string | Tag)[]) => {
```

Comment:
no need to do it in this PR, just fyi, the only reason most of these event emitter names started with `_` was because in a few cases there was no body, and the compiler will complain of an unused parameter, so it was to suppress the linter/compiler. 

As we rewrite stuff if you encounter them in other places as you touch the code just remove it.

There is no need for immediate action right now, but in general you can remove the `_` as you encounter them.